### PR TITLE
Fix matplotlib version issues and a circular import issue

### PR DESCRIPTION
**Pull request by chansigit**
- Upgrade matplotlib from 3.5.3 to 3.7.5 (we can actually move forward to 3.9.0 by abandoning python 3.8 )
- Fix a circular import issue in dym.pp.cell_cycle (this function uses dyn.tl.utils when it is partially imported), I now move imports inside functions `group_corr` and `group_score`
- I also removed python 3.7 tests. Python 3.7 is so old and doesn't support matplotlib3.7.5 and other new version scverse packages.

### CODE REVIEW

**Comment by chansigit** [comment]
@Sichao25  Hi, Sichao, I believe a lot of failed tests were attributed to [this issue](https://github.com/aristoteleo/dynamo-release/issues/687#issue-2389928227 ). 

Can we find a better way to store the data? It seems that dropbox banned the non-browser downloading? 

One possible solution is to use some professional research data sharing platform like figshare (I've uploaded the data here https://figshare.com/articles/dataset/Dynamo_sample_datasets/25690782).

**Comment by Sichao25** [comment]
> @Sichao25 Hi, Sichao, I believe a lot of failed tests were attributed to [this issue](https://github.com/aristoteleo/dynamo-release/issues/687#issue-2389928227).
> 
> Can we find a better way to store the data? It seems that dropbox banned the non-browser downloading?
> 
> One possible solution is to use some professional research data sharing platform like figshare (I've uploaded the data here https://figshare.com/articles/dataset/Dynamo_sample_datasets/25690782).
> 
> This works
> 
> ```
> adata=dyn.sample_data.get_adata(url='https://figshare.com/ndownloader/files/47420257', filename='zebrafish.h5ad')
> ```
Sounds great. Thank you for detecting this issue.

**Comment by chansigit** [comment]
> > @Sichao25 Hi, Sichao, I believe a lot of failed tests were attributed to [this issue](https://github.com/aristoteleo/dynamo-release/issues/687#issue-2389928227).
> > Can we find a better way to store the data? It seems that dropbox banned the non-browser downloading?
> > One possible solution is to use some professional research data sharing platform like figshare (I've uploaded the data here https://figshare.com/articles/dataset/Dynamo_sample_datasets/25690782).
> > This works
> > ```
> > adata=dyn.sample_data.get_adata(url='https://figshare.com/ndownloader/files/47420257', filename='zebrafish.h5ad')
> > ```
> 
> Sounds great. Thank you for detecting this issue.

I've already uploaded all dropbox-based data to figshare and changed the urls in the codes in my own branch. Now trying to pass the github action unit tests.

**Comment by Xiaojieqiu** [comment]
@chansigit excellent work! can we also support python 3.10 and 3.11, etc?

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
```
Comment (Xiaojieqiu) [inline .github/workflows/python-package.yml:18]: can we add 3.9, 3.10, 3.11, 3.12?

File: dynamo/configuration.py [hunk 2]
```diff
@@ -432,25 +443,25 @@ def update_data_store_mode(mode: str) -> None:
 with warnings.catch_warnings():
     warnings.simplefilter("ignore")
     if "zebrafish" not in matplotlib.colormaps():
-        plt.register_cmap("zebrafish", zebrafish_cmap)
+        matplotlib.colormaps.register(name="zebrafish", cmap=zebrafish_cmap)
```
Comment (Xiaojieqiu) [inline dynamo/configuration.py:446]: can we use mpl here and the same for the rest?

File: dynamo/external/scifate.py [hunk 3]
```diff
@@ -58,11 +58,11 @@ def scifate_glmnet(
         motif_ref: The path to the TF binding motif data as described above. It provides the list of TFs gene names and
             is used to process adata object to generate the TF expression and target new expression matrix for glmnet
             based TF-target synthesis rate linkage analysis. But currently it is not used for motif based filtering.
-            By default, it is a dropbox link that store the data from us. Other motif reference can bed downloaded from
+            By default, it is a cloud link that store the data from us. Other motif reference can bed downloaded from
```
Comment (Xiaojieqiu) [inline dynamo/external/scifate.py:61]: can we write figshare instead of cloud?

File: dynamo/plot/topography.py [hunk 4]
```diff
@@ -323,7 +321,7 @@ def plot_fixed_points_2d(
     if ax is None:
         ax = plt.gca()
 
-    cm = matplotlib.cm.get_cmap(_cmap) if type(_cmap) is str else _cmap
+    cm = matplotlib.colormaps[_cmap] if type(_cmap) is str else _cmap
```
Comment (Xiaojieqiu) [inline dynamo/plot/topography.py:324]: use mpl?

**Review by Xiaojieqiu — COMMENTED** [review]
I have made a few comments

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (chansigit) [reply .github/workflows/python-package.yml:18]: 3.12 failed due to the removal of `distutils` package https://stackoverflow.com/questions/77247893/modulenotfounderror-no-module-named-distutils-in-python-3-12

other versions on the way

File: dynamo/configuration.py (hunk 2 above)
Comment (chansigit) [reply dynamo/configuration.py]: done

File: dynamo/external/scifate.py (hunk 3 above)
Comment (chansigit) [reply dynamo/external/scifate.py]: done

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (chansigit) [reply .github/workflows/python-package.yml:18]: 3.10 and 3.11 also failed. we need more time to get it pass.
I noticed scanpy is asking a python version <3.10, we may remain in 3.9 temporarily.
https://github.com/scverse/scanpy/blob/db2118e8eb60bbf6287ce1413477480e16b2508b/pyproject.toml#L70

File: dynamo/plot/topography.py (hunk 4 above)
Comment (chansigit) [reply dynamo/plot/topography.py]: done

**Comment by Xiaojieqiu** [comment]
@Sichao25 can you please reviewer it and finalize it? I will merge it to the main branch later

**Review by Sichao25 — APPROVED** [review]
Looks good to me!

**Pull request merged**